daemon/main.go

```diff
@@ -117,8 +118,24 @@ var (
 )
 
 var (
-	logOpts               = make(map[string]string)
-	kvStoreOpts           = make(map[string]string)
+	logOpts                = make(map[string]string)
+	kvStoreOpts            = make(map[string]string)
+	fixedIdentity          = make(map[string]string)
+	fixedIdentityValidator = option.Validator(func(val string) (string, error) {
+		vals := strings.Split(val, "=")
+		if len(vals) != 2 {
+			return "", fmt.Errorf(`invalid fixed identity: expecting "<numeric-identity>=<identity-name>" got %q`, val)
+		}
+		ni, err := identity.ParseNumericIdentity(vals[0])
+		if err != nil {
+			return "", fmt.Errorf(`invalid numeric identity %q: %s`, val, err)
+		}
+		if !identity.IsUserReservedIdentity(ni) {
+			return "", fmt.Errorf(`invalid numeric identity %q: valid numeric identity is between %d and %d`,
+				val, identity.UserReservedNumericIdentity.Uint32(), identity.MinimalNumericIdentity.Uint32())
+		}
```

Comment:
We should validate `vals[1]` to ensure it is not equal to one of the reserved label names, e.g. host, world, ...